src/core/HorizontalTabs/HorizontalTabs.tsx

```diff
@@ -81,21 +82,26 @@ export const HorizontalTabs = (props: HorizontalTabsProps) => {
 
   const tablistRef = React.useRef<HTMLUListElement>(null);
 
+  const tablistSize = useResizeObserver(tablistRef);
+
   const [currentIndex, setCurrentIndex] = React.useState(getValidIndex());
   const [stripeStyle, setStripeStyle] = React.useState<React.CSSProperties>({});
 
   React.useLayoutEffect(() => {
     if (activeIndex != null && currentIndex !== activeIndex) {
       setCurrentIndex(getValidIndex());
     }
-    const activeTab = tablistRef.current?.children[currentIndex] as HTMLElement;
+  }, [activeIndex, currentIndex, getValidIndex]);
+
+  React.useLayoutEffect(() => {
     if (type !== 'default') {
+      const activeTab = tablistRef.current?.children[currentIndex];
       setStripeStyle({
         width: activeTab?.getBoundingClientRect().width,
-        left: activeTab?.offsetLeft,
+        left: (activeTab as HTMLElement)?.offsetLeft,
```

Comment:
I see `ResizeObserver` also return `left`/`x` prop but I don't know if there is a big difference using that or `offsetLeft`.